JASP-Desktop/components/JASP/Controls/VariablesForm.qml

```diff
@@ -23,15 +23,13 @@ import JASP				1.0
 Item
 {
 	id					: variablesForm
-	implicitWidth		: form.width
-	implicitHeight		: jaspTheme.defaultVariablesFormHeight
-	height				: implicitHeight
-	width				: implicitWidth
+	height				: preferredHeight
+	width				: preferredWidth
 	Layout.columnSpan	: parent.columns
 	visible				: !debug || DEBUG_MODE
 
 	default property alias	content:						items.children
-			property int	listWidth:						implicitWidth * 2 / 5
+			property int	listWidth:						preferredWidth * 2 / 5
```

Comment:
And this I am less sure about, but shouldn't the listWidth be based on the actual width?

Reply:
Yes, that's true it should be width in place of preferredWidth: inside a GrigLayout width is set to preferredWidth, but without GridLayout preferredWidth has no sense, and only width should be used.

Reply:
Because you never know where it might be used in the future and [the documentation](https://doc.qt.io/qt-5/qml-qtquick-item.html#implicitWidth-prop) states that:
>Defines the natural width or height of the Item if no width or height is specified.
>The default implicit size for most items is 0x0, however some items have an inherent implicit size which cannot be overridden, for example, Image and Text.
>Setting the implicit size is useful for defining components that have a preferred size based on their content,...

I can see situations where you would want to be able to select several variables in something that isn't a form. For defining computed columns from several columns for instance.
Or maybe we could replace the columnlist in the filterconstructor by this or who knows.

Reply:
Ik zet nu ook de implicitWidth/Height

Reply:
Hi Joris, I did a push force, you cannot see the changes. What I did is just to set the implicitHeight/Width.

Reply:
I can actually see the changes!
But...
Why are you setting implicitWidth/Height from preferredWidth/Height and not the other way round? Like I suggested...

Reply:
Because if the QML form changes the preferredHeight property, then the height and implicitHeight won't be changed.

Reply:
Yes, that is exactly right. It shouldn't!
implicitWidth and height must be set by the component itself, so the code we are debating right now.
It is the "natural size" of the element.

If the user wants a different size then that's fine, but then they should change width/height.
However, we want to work with layout and *just to make it easier for the users* we added preferredWidth/Height as a replacement for Layout.preferredWidth etc.

Reply:
And height will be changed by Layout.

Reply:
OK, if I set height to be preferredHeight, it solves also this issue.
Is this what you meant?

Reply:
I think I start to understand what you mean...
So the height and preferredHeight is set to implicitHeight (which has a default height). And the user has to set the preferredHeight if in Layout (that should set the right height), or directly the height if not in a layout.
I solve the issue by adding a binding.
Maybe my slow brain is by-effect of the corona virus...